comp/core/workloadmeta/collectors/internal/nvml/nvml.go

```diff
@@ -0,0 +1,123 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+package nvml
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/fx"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/config/env"
+	"github.com/DataDog/datadog-agent/pkg/errors"
+)
+
+const (
+	collectorID   = "nvml"
+	componentName = "workloadmeta-nvml"
+	nvidiaVendor  = "nvidia"
```

Comment:
nit: maybe introduce a new type alias for the vendor (for future expansion once we start adding more vendors)?